apps/mysql.go

```diff
@@ -15,13 +15,65 @@
 package apps
 
 import (
-	"fmt"
-	"strings"
-
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+
+	"fmt"
+	"strings"
 )
 
+type MetricsReceiverMySQL struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+
+	confgenerator.MetricsReceiverShared `yaml:",inline"`
+
+	Endpoint string `yaml:"endpoint" validate:"omitempty,url"`
+
+	Password string `yaml:"password" validate:"omitempty"`
+
+	Username string `yaml:"username" validate:"omitempty"`
+}
+
+const defaultEndpoint = "localhost:3306"
```

Comment:
I do not think we should default to a TCP connection. Especially on distros that automatically configure Unix socket authentication, we should make sure we are using a Unix socket so the zero-config setup works out of the box.

Reply:
I'll update this to detect the default socket file from `/var/run/mysqld/mysql.sock` once the upstream PR gets Merged.

Reply:
open-telemetry/opentelemetry-collector-contrib#6138 was merged.